### ISSUE TITLE
ESLint: Cannot read property "name" of null on next/link component

### ISSUE THREAD

**Issue by noaione** [vercel/next.js#26412]
### What version of Next.js are you using?

11.0.0

### What version of Node.js are you using?

14.16.0

### What browser are you using?

Chrome 91

### What operating system are you using?

Windows

### How are you deploying your application?

None

### Describe the Bug

When I try to run eslint, it crashed while trying to lint `next/link`.

```
TypeError: Cannot read property 'name' of null
Occurred while linting T:\Coding\GitHub\blog.n4o.xyz\components\Link.js:15
    at T:\Coding\GitHub\blog.n4o.xyz\node_modules\@next\eslint-plugin-next\lib\rules\link-passhref.js:49:33
    at Array.some (<anonymous>)
    at JSXOpeningElement (T:\Coding\GitHub\blog.n4o.xyz\node_modules\@next\eslint-plugin-next\lib\rules\link-passhref.js:45:41)
    at T:\Coding\GitHub\blog.n4o.xyz\node_modules\eslint\lib\linter\safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (T:\Coding\GitHub\blog.n4o.xyz\node_modules\eslint\lib\linter\safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (T:\Coding\GitHub\blog.n4o.xyz\node_modules\eslint\lib\linter\node-event-generator.js:293:26)
    at NodeEventGenerator.applySelectors (T:\Coding\GitHub\blog.n4o.xyz\node_modules\eslint\lib\linter\node-event-generator.js:322:22)
    at NodeEventGenerator.enterNode (T:\Coding\GitHub\blog.n4o.xyz\node_modules\eslint\lib\linter\node-event-generator.js:336:14)
    at CodePathAnalyzer.enterNode (T:\Coding\GitHub\blog.n4o.xyz\node_modules\eslint\lib\linter\code-path-analysis\code-path-analyzer.js:711:23)
```

Even with this .eslint config, it still failed:
```js
module.exports = {
  extends: ['next'],
}
```

The code is here: https://github.com/noaione/blog.n4o.xyz/blob/master/components/Link.js#L15
**Version**
- ESLint: 7.29.0
- eslint-config-next: 11.0.0

### Expected Behavior

ESlint doesn't crash, and report what should be fixed or not.

### To Reproduce

The GitHub repository is here: https://github.com/noaione/blog.n4o.xyz
Currently, just having a `.eslintrc.js` like this
```js
module.exports = {
  extends: ['next'],
}
```
Makes ESLint crash.
If I ignore the `components/Link.js` file, it will run just fine.

**Comment by noaione** [comment]
Apparently, this has been fixed in canary version (`11.0.1-canary.5` and up). I should have check that first before opening a new issue

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
